app/src/main/java/ca/unb/mobiledev/reflexrevolution/activities/GameOverActivity.java

```diff
@@ -58,4 +68,19 @@ protected void onCreate(Bundle savedInstanceState) {
         // Menu button to close this activity, returning to game select screen
         menuButton.setOnClickListener(v -> finish());
     }
+
+    private void initSharedPreferences() {
+        prefs = getSharedPreferences(PREFS_FILE_NAME, Context.MODE_PRIVATE);
+    }
+
+    private int readHighScoreFromSharedPreferences() {
+        return prefs.getInt(HIGH_SCORE_KEY, 0);
+    }
+
+    private void writeHighScoreToSharedPreferences(int score) {
+        SharedPreferences.Editor editor = prefs.edit();
+        editor.putInt(HIGH_SCORE_KEY, score);
+        editor.apply();
+    }
```

Comment:
It might be better to have these methods in their own separate static class. That way we can access the saved data from any activity very easily.

For example, we could have the static class like this:
```java
public class LocalData {
    private static final String PREFS_FILE_NAME = "AppPrefs";
    private static final String HIGH_SCORE_KEY = "HIGH_SCORE_KEY";
    private static SharedPreferences prefs;

    public static void initialize(Context context) {
        prefs = context.getSharedPreferences(PREFS_FILE_NAME, Context.MODE_PRIVATE);
    }

    public static int getHighScore() {
        return prefs.getInt(HIGH_SCORE_KEY, 0);
    }

    public static void setHighScore(int score) {
        SharedPreferences.Editor editor = prefs.edit();
        editor.putInt(HIGH_SCORE_KEY, score);
        editor.apply();
    }
}
```

So any activity can use it like so:
```java
LocalData.initialize(this);
int highScore = LocalData.getHighScore();
LocalData.setHighScore(100);
```

Reply:
Also note that the high score is unique for each gamemode and difficulty level, so we would need to pass that information in when getting/setting the high score. We can use the names of the gamemode & difficulty to create the specific key for each combination. Something like this:
```java
public static int getHighScore(GameMode gameMode, Difficulty difficulty) {
    String key = HIGH_SCORE_KEY+"_"+gameMode.getName()+"_"+difficulty.getName();
    return prefs.getInt(key, 0);
}
```

Reply:
I think Fred's suggestions seem like a great way to do it. Only thing I would add is maybe you would also have a private static method to get they key based on GameMode and Difficulty, which can then be re-used in the getHighScore() and setHighScore() methods. So like:

`private static int getHighScoreKey(GameMode gameMode, Difficulty difficulty) {
    return "HIGH_SCORE_KEY_"+gameMode.getName()+"_"+difficulty.getName();
}

public static int getHighScore(GameMode gameMode, Difficulty difficulty) {
    return prefs.getInt(getHighScoreKey(gameMode, difficulty), 0);
}

public static void setHighScore(int score, GameMode gameMode, Difficulty difficulty) {
        SharedPreferences.Editor editor = prefs.edit();
        editor.putInt(getHighScoreKey(gameMode, difficulty), score);
        editor.apply();
    }
`